src/touch_save_app.c

```diff
@@ -0,0 +1,520 @@
+    InitBgFromTemplate(bgConfig, 6, &ov30_0225DC64, GF_BG_TYPE_TEXT);
+    BG_ClearCharDataRange(6, 0x20, 0, HEAP_ID_4);
+    BgClearTilemapBufferAndCommit(bgConfig, 6);
+    BgClearTilemapBufferAndCommit(bgConfig, 5);
+
+    GXS_SetVisibleWnd(0);
+    GX_EngineBToggleLayers(1, GX_LAYER_TOGGLE_OFF);
+    GX_EngineBToggleLayers(2, GX_LAYER_TOGGLE_OFF);
+    GX_EngineBToggleLayers(4, GX_LAYER_TOGGLE_OFF);
+    GX_EngineBToggleLayers(16, GX_LAYER_TOGGLE_OFF);
+
+    SysTask *task = sub_02007200((SysTaskFunc)ov30_0225D700, sizeof(TouchSaveAppData), 10, HEAP_ID_8);
+    TouchSaveAppData *data = sub_0201F988(task);
+    data->task = task;
+    data->unk0 = 0;
+    data->bgConfig = bgConfig;
+    data->unk18 = a1;
+    data->fieldSystem = fieldSystem;
+    data->unk30 = 0;
+    data->options = Save_PlayerData_GetOptionsAddr(data->fieldSystem->saveData);
+    data->state = TOUCHSAVEAPP_STATE_DISPLAY_SAVE_INFORMATION;
+
+    TouchSaveApp_SetupGraphics(data);
+    TouchSaveApp_SetupText(data, data->fieldSystem->unkD2_0);
+
+    GX_EngineBToggleLayers(1, GX_LAYER_TOGGLE_ON);
+    GX_EngineBToggleLayers(2, GX_LAYER_TOGGLE_ON);
+    GX_EngineBToggleLayers(4, GX_LAYER_TOGGLE_ON);
+    GX_EngineBToggleLayers(16, GX_LAYER_TOGGLE_ON);
+
+    sub_02002B8C(TRUE);
+
+    return task;
+}
+
+void ov30_0225D64C(BgConfig *bgConfig, SysTask *task) {
+    TouchSaveAppData *data = sub_0201F988(task);
+
+    FS_LoadOverlay(MI_PROCESSOR_ARM9, FS_OVERLAY_ID(OVY_123));
+
+    if (!ov123_0225F4A8(ov30_0225DC28)) {
+        AllocFromHeapAtEnd(HEAP_ID_3, 1000);
+    }
+
+    ov01_021F434C(data->unk40);
+    ov01_021F43D0(data->unk40);
+
+    sub_02002B8C(FALSE);
+
+    if (ov123_0225F520(ov30_0225DC08)) {
+        AllocFromHeapAtEnd(HEAP_ID_3, 1000);
+    }
+
+    TouchSaveApp_DestroyWindow(data);
+    TouchSaveApp_DestroyText(data);
+
+    sub_02007234(task);
+
+    FreeBgTilemapBuffer(bgConfig, 6);
+    FreeBgTilemapBuffer(bgConfig, 5);
+    FreeBgTilemapBuffer(bgConfig, 4);
+
+    DestroyHeap(HEAP_ID_8);
+
+    if (ov123_0225F610(ov30_0225DC18)) {
+        AllocFromHeapAtEnd(HEAP_ID_3, 1000);
+    }
+
+    FS_UnloadOverlay(MI_PROCESSOR_ARM9, FS_OVERLAY_ID(OVY_123));
+}
+
+BOOL ov30_0225D6FC(void *a0) {
+    return TRUE;
+}
+
+static void ov30_0225D700(SysTask *task, TouchSaveAppData *data) {
+    if (!FieldSystem_TaskIsRunning(data->fieldSystem)) {
+        if (data->fieldSystem->unkD2_0 == 2) {
+            data->fieldSystem->unkD2_0 = 3;
+            ov01_021F434C(data->unk40);
+        }
+
+        G2S_BlendNone();
+    } else {
+        if (data->fieldSystem->unkD2_0 == 2 && sTouchSaveAppStateFuncs[data->state](data) == TRUE) {
+            data->fieldSystem->unkD2_0 = 3;
+            ov01_021F6A9C(data->fieldSystem, 0, 0);
+        }
+    }
+}
+
+static void TouchSaveApp_SetupGraphics(TouchSaveAppData *data) {
+    G2S_BlendNone();
+
+    GfGfxLoader_LoadCharData(NARC_a_0_1_4, 76, data->bgConfig, GF_BG_LYR_SUB_0, 0, 0, TRUE, HEAP_ID_8);
```

Comment:
ideally use the naix constants, not sure if a narcs can do that

Reply:
You can if you dump the NARC, however I have no idea what to name it or where to put it